src/bundle/Core/Command/ExpireUserPasswordsCommand.php

```diff
@@ -0,0 +1,383 @@
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\ContentList;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\LogicalAnd;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\API\Repository\Values\ContentType\FieldDefinition;
+use eZ\Publish\API\Repository\Values\Filter\Filter;
+use eZ\Publish\SPI\Persistence\User\Handler;
+use InvalidArgumentException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ExpireUserPasswordsCommand extends Command
+{
+    protected static $defaultName = 'ibexa:user:expire-password';
+
+    public const DEFAULT_BATCH_SIZE = 50;
+
+    public const DEFAULT_PASSWORD_TTL = 00;
+
+    public const BEFORE_RUNNING_HINTS = <<<EOT
+<error>Before you continue:</error>
+- Make sure to back up your database.
+- Take installation offline, during the script execution the database should not be modified.
+- Run this command without memory limit.
+- Run this command in production environment using <info>--env=prod</info>
+EOT;
+
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
+
+    /** @var \eZ\Publish\SPI\Persistence\User\Handler */
+    private $userHandler;
+
+    public function __construct(
+        Repository $repository,
+        Handler $userHandler
+    ) {
+        $this->repository = $repository;
+        $this->userHandler = $userHandler;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $beforeRunningHints = self::BEFORE_RUNNING_HINTS;
+        $this
+            ->setDescription('Expire passwords for selected users.')
+            ->addOption(
+                'user-id',
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                'Expire password for specific User identified by ID'
+            )
+            ->addOption(
+                'user-group-id',
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                'Expire passwords of all users assigned to specific User Group'
+            )
+            ->addOption(
+                'user-content-type-identifier',
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                'Expire passwords of all users based on specific Content Type'
+            )
+            ->addOption(
+                'force',
+                null,
+                InputOption::VALUE_NONE,
+                'Perform setting user passwords as expired'
+            )
+            ->addOption(
+                'batch-size',
+                null,
```

Comment:
Convention: multiple commands in this bundle for such thing use `iteration-count` and `c` shortcut.